server/interaction/npc.go

```diff
@@ -0,0 +1,132 @@
+package interaction
+
+import (
+	"github.com/joshprzybyszewski/cribbage/game"
+	"github.com/joshprzybyszewski/cribbage/logic/scorer"
+	"github.com/joshprzybyszewski/cribbage/model"
+	"github.com/joshprzybyszewski/cribbage/utils/rand"
+)
+
+// NPCType is an enum specifying which type of NPCType
+type NPCType int
+
+// Dumb, Simple, and Calculated are supported
+const (
+	Dumb NPCType = iota
+	Simple
+	Calculated
+)
+
+var npcs = [...]npcPlayer{
+	Dumb: {
+		Type: Dumb,
+		id:   `dumbNPC`,
+	},
+	Simple: {
+		Type: Simple,
+		id:   `simpleNPC`,
+	},
+	Calculated: {
+		Type: Calculated,
+		id:   `calculatedNPC`,
+	},
+}
+
+var _ Player = (*npcPlayer)(nil)
+
+type npcPlayer struct {
+	Type NPCType
```

Comment:
🌵 if we can avoid the field name `Type`, I would prefer it since `type` is a keyword in golang